bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/OpenBuilder.java

```diff
@@ -0,0 +1,39 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *pa
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.bookkeeper.client.api;
+
+import java.util.concurrent.CompletableFuture;
+import org.apache.bookkeeper.client.BKException;
+
+/**
+ * Builder-style interface to open exiting ledgers
+ */
+public interface OpenBuilder extends OpBuilder<ReadHandle> {
+
+    OpenBuilder withLedgerId(long ledgerId);
+
+    OpenBuilder withRecovery(boolean recovery);
+
+    OpenBuilder withPassword(byte[] password);
+
+    OpenBuilder withDigestType(org.apache.bookkeeper.client.BookKeeper.DigestType digestType);
```

Comment:
1. why do we use a fully-qualified package name here?
2. it will be strange to have `org.apache.bookkeeper.client.api` back-reference classes in `org.apache.bookkeeper.client`. since we are introducing new client interface, I would suggest adding a DigestType in api. so the new interface only uses this new class. in the actually implementation, we can convert `api.DigestType` back to the old `DigestType`. so when we get rid of old-style client api, we don't have any troubles.

Reply:
@sijie  what about BKException ? I think this is very difficult to drop, but we can try to convert all the BKException inside the CompletableFutures

Reply:
@eolivelli - I think BKException is only used for declaring all the sync methods. so if we get rid of sync methods, we don't need to use BKException in `client.api` package. You can still provide a util function on waiting for the result of a completablefuture, and this util function still in `client` package and is part of the implementation.